qt-models/diveplannermodel.cpp

```diff
@@ -986,32 +986,29 @@ struct divedatapoint * DivePlannerPointsModel::cloneDiveplan(struct diveplan *pl
 
 int DivePlannerPointsModel::analyzeVariations(struct decostop *min, struct decostop *mid, struct decostop *max, const char *unit)
 {
+	int minsum = 0;
+	int midsum = 0;
+	int maxsum = 0;
 	int leftsum = 0;
 	int rightsum = 0;
-	while (mid->depth > min->depth)
-		++mid;
-	while (max->depth > mid->depth)
-		++max;
 
-	while (mid->depth) {
-		int left = mid->time - min->time;
-		leftsum += left;
-		int right = max->time - mid->time;
-		rightsum += right;
-#ifdef SHOWSTOPVARIATIONS
-		if (min->time + mid->time + max->time)
-			printf("%dm: %dmin + %ds/%s +- %ds/%s\n", mid->depth / 1000,
-			       (mid->time + 1)/60,
-			       (left + right) / 2, unit,
-			       (right - left) / 2, unit);
-#else
-		(void) unit;
-#endif
+	while (min->depth) {
+		minsum += min->time;
 		++min;
+	}
+	while (mid->depth) {
+		midsum += mid->time;
 		++mid;
+	}
+	while (max->depth) {
+		maxsum += max->time;
 		++max;
 	}
-#ifdef SHOWSTOPVARIATIONS
+
+	leftsum = midsum - minsum;
```

Comment:
Now, I remember, why I had the more complicated logic: At some point, I was printing the differences for all individual stops. But since we ended up only using the aggregated values, this is no longer needed.